### PR TITLE
Add support for file versions

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
- adds support for versions
- only visible for NC14 user
- they are shown in activity stream
	- you can click on "revert" icon
- currently this is not MVP, but I will open up a new PR after the other file detail PRs by @AndyScherzinger are merged

TODO
- [x] going back from detail view, reload of file list needs to be done, @AndyScherzinger do you have an idea for this?
- [x] after successful restore: current downloaded file needs to be removed
- [x] @rullzer what about old thumbnails? E.g. you have an image you change?

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
> going back from detail view, reload of file list needs to be done, @AndyScherzinger do you have an idea for this?

There is no going back ;) On a tablet it would be a split pane between file list and file details :) I also did this for the file details already by introducing a listener interface (implemented by the activity) so you could just extend that one and ping the activity that it needs to refresh the other fragment/list :)

**Comment by rullzer** [comment]
@tobiasKaminsky how do you currently evaluate if the preview is still valid? Because you could use the same logic here. Either check the etag of the preview against the endpoint. Or check the etag of the actual file that should also change.

**Comment by tobiasKaminsky** [comment]
Thanks @rullzer, currently we are checking during sync if the modification date has changed and if so, we simply fetch a new thumbnail.
As we currently do not check eTags for thumbnails, I will create this in a different branch, so that this one will not blown up.

**Comment by tobiasKaminsky** [comment]
As comment PR is based on this, I did not rebase, but merge.
As soon as this is then ready, prior merging this PR we need to rebase, to get rid of the merge commit.

**Comment by tobiasKaminsky** [comment]
This is now ready for code review & testing.

File: src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java [hunk 1]
```diff
@@ -183,6 +183,15 @@ public FileDetailSharingFragment getFileDetailSharingFragment() {
         return ((FileDetailTabAdapter)viewPager.getAdapter()).getFileDetailSharingFragment();
     }
 
+    /**
+     * return the reference to the file detail activity fragment to communicate with it.
+     *
+     * @return reference to the {@link FileDetailSharingFragment}
+     */
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java]: --> FileDetailActivitiesFragment ;)

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java]: Copy&paste error ;-)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I only get an error screen (with the icon and so on) when opening a file that has versions on your testing Nc14 server instance :/

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger can you write me on irc or per mail the exact file alongside with the credentials? Then I can have a look at exact the same file.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger this was because I cloned server manually and forgot activity app.
Please re-test it.

! I rebased it!

File: src/main/res/drawable/ic_history.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (AndyScherzinger) [inline src/main/res/drawable/ic_history.xml]: can/should we add a license header here, donb't know under which license the Nextcloud icons are published though

**Comment by AndyScherzinger** [comment]
:+1: works like a charm add @tobiasKaminsky - I also fixed any (really tiny, minor) issues I found :+1: 

Only one small question remained regarding license header of the restore icon.

File: src/main/res/drawable/ic_history.xml (hunk 2 above)
Comment (tobiasKaminsky) [reply src/main/res/drawable/ic_history.xml]: @nextcloud/designers maybe? :-)

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger thanks for your feedback, I added everything :+1:

File: src/main/res/drawable/ic_history.xml (hunk 2 above)
Comment (jancborchardt) [reply src/main/res/drawable/ic_history.xml]: AGPLv3, and if that doesn’t work CC-BY-SA :)

**Comment by mario** [comment]
👍

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky shouldn't you have merged https://github.com/nextcloud/android-library/pull/150 first and moved back to referencing master-SNAPSHOT???

**Comment by tobiasKaminsky** [comment]
Ups. 
As I did not wrote this in the first comment, I just forgot about it :/

@mario can you quickly have a look at https://github.com/nextcloud/android-library/pull/150? :-D

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky nextcloud/android-library#150 has been merged and master has been build via jitpack

**Pull request merged**